build-settings-logic/src/main/kotlin/atomicfu-cache-redirector.settings.gradle.kts

```diff
@@ -0,0 +1,202 @@
+    "https://jcenter.bintray.com" to "https://cache-redirector.jetbrains.com/jcenter",
+    "https://www.python.org/ftp" to "https://cache-redirector.jetbrains.com/www.python.org/ftp",
+    "https://registry.npmjs.org" to "https://cache-redirector.jetbrains.com/registry.npmjs.org",
+    "https://maven.google.com" to "https://cache-redirector.jetbrains.com/maven.google.com",
+    "https://dl.google.com/go" to "https://cache-redirector.jetbrains.com/dl.google.com/go",
+    "https://dl.google.com/go" to "https://cache-redirector.jetbrains.com/dl.google.com.go",
+    "https://clojars.org/repo" to "https://cache-redirector.jetbrains.com/clojars.org/repo",
+    "https://nodejs.org/dist" to "https://cache-redirector.jetbrains.com/nodejs.org/dist",
+    "https://jitpack.io" to "https://cache-redirector.jetbrains.com/jitpack.io",
+    "https://repo.labs.intellij.net/intellij-nightly-sdk" to "https://cache-redirector.jetbrains.com/www.jetbrains.com/intellij-repository/nightly",
+    "https://d1lc5k9lerg6km.cloudfront.net" to "https://cache-redirector.jetbrains.com/www.jetbrains.com/jps-cache/intellij",
+    "https://www.jetbrains.com/intellij-repository" to "https://cache-redirector.jetbrains.com/www.jetbrains.com/intellij-repository",
+    "https://www.jetbrains.com/intellij-repository" to "https://cache-redirector.jetbrains.com/intellij-repository",
+    "https://download.jetbrains.com" to "https://cache-redirector.jetbrains.com/download.jetbrains.com",
+    "https://secure.index-cdn.jetbrains.com" to "https://cache-redirector.jetbrains.com/secure.index-cdn.jetbrains.com",
+    "https://d2xrhe97vsfxuc.cloudfront.net" to "https://cache-redirector.jetbrains.com/intellij-jbr",
+    "https://d2xrhe97vsfxuc.cloudfront.net" to "https://cache-redirector.jetbrains.com/intellij-jdk",
+    "https://d2xrhe97vsfxuc.cloudfront.net" to "https://cache-redirector.jetbrains.com/jetbrains.bintray.com/intellij-jbr",
+    "https://d2xrhe97vsfxuc.cloudfront.net" to "https://cache-redirector.jetbrains.com/jetbrains.bintray.com/intellij-jdk"
+)
+
+val aliases = mapOf(
+    "https://repo.maven.apache.org/maven2" to "https://repo1.maven.org/maven2" // Maven Central
+)
+
+fun URI.maybeRedirect(): URI {
+    val url = toString().trimEnd('/')
+    val deAliasedUrl = aliases.getOrDefault(url, url)
+
+    val cacheUrlEntry = cacheMap.entries.find { (origin, _) -> deAliasedUrl.startsWith(origin) }
+    return if (cacheUrlEntry != null) {
+        val cacheUrl = cacheUrlEntry.value
+        val originRestPath = deAliasedUrl.substringAfter(cacheUrlEntry.key, "")
+        URI("$cacheUrl$originRestPath")
+    } else {
+        this
+    }
+}
+
+fun RepositoryHandler.redirect() = configureEach {
+    when (this) {
+        is MavenArtifactRepository -> url = url.maybeRedirect()
+        is IvyArtifactRepository -> @Suppress("SENSELESS_COMPARISON") if (url != null) {
+            url = url.maybeRedirect()
+        }
+    }
+}
+
+pluginManagement.repositories.redirect()
```

Comment:
@adam-enko honestly say, I still don't understand, why do you suggest to move it to gradle dir. It looks like a real settings plugin, which should be applied directly to `settings.gradle.kts`.
I would suggest not to move it.

Reply:
Well, because it's not a real settings plugin :) It's applied as a file-based script plugin in `build-settings-logic/settings.gradle.kts` and in all of the integration tests.

While technically it can be both a file-based and pre-compiled script plugin, it's a little odd. I also don't like that it's easy to accidentally break the the plugin by using a utility function from Utils.kt.

That said, if it works, it works. I'm not going to block the PR because of it.